pipeline/metadata/flatten_satellite.py

```diff
@@ -371,54 +463,60 @@ def _process_satellite_v2_control(
         Rows
     """
     responses = scan.get('response', [])
-    if responses:
-      # An overall satellite v2 measurement
-      # always contains some non-control trial domains
-      is_control_domain = False
+
+    for response in responses:
+      is_control_domain = flatten_base.is_control_url(response['url'])
 
       row = {
           'domain':
-              scan['test_url'],
+              response['url'],
           'is_control':
               is_control_domain,
           'category':
-              self.category_matcher.get_category(scan['test_url'],
+              self.category_matcher.get_category(response['url'],
                                                  is_control_domain),
           'ip':
               scan['vp'],
           'is_control_ip':
               True,
           'date':
-              responses[0]['start_time'][:10],
+              response['start_time'][:10],
           'start_time':
-              format_timestamp(responses[0]['start_time']),
+              format_timestamp(response['start_time']),
           'end_time':
-              format_timestamp(responses[-1]['end_time']),
+              format_timestamp(response['end_time']),
           'anomaly':
               None,
           'success':
               not scan['connect_error'],
           'controls_failed':
               not scan['passed_control'],
-          'rcode': [str(response['rcode']) for response in responses],
+          'rcode': [str(response['rcode'])],
```

Comment:
This should not be a list.

Reply:
That will be changed in https://github.com/censoredplanet/censoredplanet-analysis/pull/117, but not this PR.